vispy/visuals/axis.py

```diff
@@ -0,0 +1,93 @@
+
+import numpy as np
+
+from .visual import Visual
+from .line import LineVisual
+from .text import TextVisual
+
+class AxisVisual(Visual):
+    """Axis visual
+
+    Parameters
+    ----------
+    extents : array
+        Co-ordinates of start and end of the axis.
+    tick_direction : array
+        Unit vector specifying the direction of the ticks. Default is
+        left/down. (This will be normalised to a unit vector)
+    domain : tuple
+        The data values at the beginning and end of the axis, used for tick
+        labels. i.e. (5, 10) means the axis starts at 5 and ends at 10.
+    scale_type : str
+        The type of scale. Default is 'linear'. Options:
+            * "linear"
+            * "logarithmic"
+            * "power"
+    """
+    def __init__(self, extents, tick_direction = None, domain=(0., 1.),
+                 scale_type="linear", axis_color=(1, 1, 1, 1),
+                 tick_color = (0.7, 0.7, 0.7, 1)):
+        self.axis_color = axis_color
+        self.tick_color = tick_color
+
+        self.extents = extents
+        self.vec = np.subtract(self.extents[1],self.extents[0])
+
+        self.tick_direction = tick_direction or self._get_tick_direction()
+
+    def draw(self, transforms):
+        v_line = LineVisual(pos=self.extents, color=self.axis_color,
+                          mode='gl', width=3.0)
+
+        tick_fractions, tick_labels = self._get_tick_frac_labels()
+
+        tick_pos, tick_label_pos = self._get_tick_positions(tick_fractions)
+
+        v_ticks = LineVisual(pos=tick_pos, color=self.tick_color, mode='gl',
+                           width=2.0, connect='segments')
+
+        # TODO: make this a more efficient iterator
+        for x in range(0, len(tick_labels)):
+            TextVisual(str(tick_labels[x]), font_size=8, color='w',
+                       pos=tick_label_pos[x]).draw(transforms)
```

Comment:
It could be the creation of all of these `TextVisual` instances, you should create them once at the top in a list and then call `.draw` on them here instead of creating them again and again